pkg/kubelet/mirror_manager.go

```diff
@@ -123,16 +118,90 @@ func isMirrorPod(pod *api.Pod) bool {
 
 // This function separate the mirror pods from regular pods to
 // facilitate pods syncing and mirror pod creation/deletion.
-func filterAndCategorizePods(pods []api.Pod) ([]api.Pod, util.StringSet) {
+func filterAndCategorizePods(pods []api.Pod) ([]api.Pod, mirrorPods) {
 	filteredPods := []api.Pod{}
-	mirrorPods := util.NewStringSet()
+	mirrorPods := newMirrorPods()
+
 	for _, pod := range pods {
-		name := GetPodFullName(&pod)
-		if isMirrorPod(&pod) {
-			mirrorPods.Insert(name)
-		} else {
+		mirrorPods.Insert(&pod)
+		if !isMirrorPod(&pod) {
 			filteredPods = append(filteredPods, pod)
 		}
 	}
-	return filteredPods, mirrorPods
+	return filteredPods, *mirrorPods
+}
+
+// This is not thread-safe to use alone.
```

Comment:
nit: This is "thread-compatible". Internally we commonly use: thread-safe, thread-compatible, and thread-hostile. This article explains the definition of each: http://www.ibm.com/developerworks/library/j-jtp09263/